pkg/apb/rhcc_registry.go

```diff
@@ -63,19 +63,73 @@ func (r RHCCRegistry) LoadSpecs() ([]*Spec, int, error) {
 func (r RHCCRegistry) imageToSpec(image *Image) (*Spec, error) {
 	r.log.Debug("RHCCRegistry::imageToSpec")
 	_spec := &Spec{}
-	// Setting ID to foo because Dao expects non blank IDs
-	_spec.Id = "foo_id"
-	var name = strings.SplitAfter(image.Name, "/")
-	_spec.Name = name[len(name)-1]
-	_spec.Image = image.Name
-	_spec.Description = image.Description
+
+	req, err := http.NewRequest("GET", r.config.Url+"/v2/"+image.Name+"manifests/latest", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	type label struct {
```

Comment:
Part of this is in ```pkg/apb/types/go``` https://github.com/openshift/ansible-service-broker/blob/master/pkg/apb/types.go#L15.

The way you're reading in the json IMO is better.  Right now we use the string 'com.redhat.apb.spec' as a map key when we should really be decoding into a struct. We could either leave this as additional tech debt or try and fix it now. I might take a crack at doing this tomorrow.